cmake/kokkos_enable_devices.cmake

```diff
@@ -25,6 +25,16 @@ IF (KOKKOS_ENABLE_PTHREAD)
   SET(KOKKOS_ENABLE_THREADS ON)
 ENDIF()
 
+# as CMAKE_CXX_SIMULATE_ID does not work, detect clang-cl to avoid clang++,
+# cl, and clang-cl clashes, requires CMake >= 3.15
```

Comment:
This doesn't work we have lower cmake version as minimum

Reply:
I had this in mind. For a lower-version cmake, it will default to clang-cl. Users that specifically use clang++ need CMake >= 3.15.